gnome-initial-setup/pages/language/gis-language-page.c

```diff
@@ -126,6 +126,9 @@ language_changed (CcLanguageChooser  *chooser,
   setlocale (LC_MESSAGES, priv->new_locale_id);
   gtk_widget_set_default_direction (gtk_get_locale_direction ());
 
+  /* gis spawns processes that also need to be localised */
+  g_setenv ("LC_MESSAGES", priv->new_locale_id, TRUE);
```

Comment:
Hmm... I see we've been doing this since 2014, and it's written by Igalia (whoops), but using setenv() like this is not threadsafe, and it's known to cause random crashes if getenv() is called in another thread, which in practice happens all the time. We actually had a bug a couple years ago where code like this in gnome-session caused a very rare crash that brought down the entire desktop session. So this is not OK. Instead, we should take care to manually set LC_MESSAGES in the envp whenever gnome-initial-setup spawns a helper process. If you don't have time to fix that now, I think it would suffice to file a Phabricator issue. We just don't want to forget to fix this.